### PR TITLE
Linter: Fix dynamic errors and function complexity

### PR DESCRIPTION
**Pull request by asmaloney**
Error texts were also slightly modified so we could just use one error instead of defining several.

Complexity was fixed by replacing a 2-iteration loop with one function call (removing spaces from a string).

### CODE REVIEW

**Review by gucio321 — APPROVED** [review]
cool

File: CSS.go [hunk 1]
```diff
@@ -22,6 +22,22 @@ import (
...
+func (e ErrParse) Error() string {
+	errStr := fmt.Sprintf("unable to parse %s: %q", e.What, e.Value)
+
+	if e.Detail != nil {
+		errStr += fmt.Sprintf(" - %s", e.Detail.Error())
+	}
+
+	return errStr
+}
+
```
[the remark this answers is not in the document]
Comment (asmaloney) [reply CSS.go:40]: This has the advantage of being able to check for specific types of error as well using:

```go
errors.Is(err, ErrParse)
```

**Comment by asmaloney** [comment]
I have no idea how to read those Codecov things. I never use it.

> will decrease coverage by 0.00%.

Is it actually useful?

File: CSS.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gucio321) [reply CSS.go:40]: so, maybe rename it to `ErrCSSParse`?

[the remark this answers is not in the document]
Comment (asmaloney) [reply CSS.go:40]: Sure!

**Comment by gucio321** [comment]
idk, I rememember that I was playing with this codecov stuff some time ago.
It is useful in projects that has _decent_ test coverage, but for giu (that, since being a C wrapper) it doesn't seem useful.
At least as long as someone develops a way to write tests for our widgets.

**Comment by asmaloney** [comment]
👍 I would suggest disabling codecov then. It's just noise.

I would also suggest enabling "Test and coverage" and "golangci-lint" as **required** to pass now.

**Comment by gucio321** [comment]
> :+1: I would suggest disabling codecov then. It's just noise.
I'd turn off only these comments, but idk how :smile:
I suppose it should be done by @AllenDang somewhere on codecov's cotrol panel.


> I would also suggest enabling "Test and coverage" and "golangci-lint" as required to pass now.
"Test and coverage" is just for codecov, but golangci-lint could be required ofc ;-)

**Comment by asmaloney** [comment]
> "Test and coverage" is just for codecov

Ah - right. That _is_ the codecov. 😄  I though I saw some tests in there that were being run, but alas there are none.

`go test` is actually run anyways in the build.yml which is fine if tests are added in the future

So I think we just need to remove _coverage.yaml_ from the workflows?

**Comment by asmaloney** [comment]
>  I suppose it should be done by @AllenDang somewhere on codecov's cotrol panel.

Yes. They don't make it very easy.

Ah here it is: **Require status checks to pass before merging** in the _Settings/Branches/Branch protection rules_.

Looks like you have to list them all? Not a great UX!

**Comment by gucio321** [comment]
so its just necessary to add golangci-lint there, since build.yaml is `required` atm.

**Comment by asmaloney** [comment]
From what I read, if you turn it on, you need to add them all (_build / Building (ubuntu-latest)_, _build / Building (ubuntu-latest)_, _build / Building (ubuntu-latest)_, and _golangci-lint / Lint_).

When I look at another one of my repos, it explicitly lists each item in the matrix...

**Comment by gucio321** [comment]
yah, but `build` is there already:
image

**Pull request merged**